central/graphql/resolvers/image_vulnerabilities.go

```diff
@@ -101,3 +212,323 @@ func withImageCveTypeFiltering(q string) string {
 	return search.AddRawQueriesAsConjunction(q,
 		search.NewQueryBuilder().AddExactMatches(search.CVEType, storage.CVE_IMAGE_CVE.String()).Query())
 }
+
+func (resolver *imageCVEResolver) withImageVulnerabilityScope(ctx context.Context) context.Context {
+	return scoped.Context(ctx, scoped.Scope{
+		ID:    resolver.data.GetId(),
+		Level: v1.SearchCategory_IMAGE_VULNERABILITIES,
+	})
+}
+
+func (resolver *imageCVEResolver) getImageCVEQuery() *v1.Query {
+	return search.NewQueryBuilder().AddExactMatches(search.CVEID, resolver.data.GetId()).ProtoQuery()
+}
+
+func (resolver *imageCVEResolver) getImageCVERawQuery() string {
+	return search.NewQueryBuilder().AddExactMatches(search.CVEID, resolver.data.GetId()).Query()
+}
+
+func (resolver *imageCVEResolver) EnvImpact(ctx context.Context) (float64, error) {
+	allCount, err := resolver.root.DeploymentCount(ctx, RawQuery{})
+	if err != nil || allCount == 0 {
+		return 0, err
+	}
+	scopedCount, err := resolver.root.DeploymentCount(resolver.withImageVulnerabilityScope(ctx), RawQuery{})
+	if err != nil {
+		return 0, err
+	}
+	return float64(scopedCount) / float64(allCount), nil
+}
+
+func (resolver *imageCVEResolver) FixedByVersion(ctx context.Context) (string, error) {
+	scope, hasScope := scoped.GetScope(ctx)
+	if !hasScope {
+		return "", nil
+	}
+	if scope.Level != v1.SearchCategory_IMAGE_COMPONENTS {
+		return "", nil
+	}
+
+	edgeID := postgres.IDFromPks([]string{scope.ID, resolver.data.GetId()})
+	edge, found, err := resolver.root.ComponentCVEEdgeDataStore.Get(ctx, edgeID)
+	if err != nil || !found {
+		return "", err
+	}
+	return edge.GetFixedBy(), nil
+}
+
+func (resolver *imageCVEResolver) IsFixable(ctx context.Context, args RawQuery) (bool, error) {
+	query, err := args.AsV1QueryOrEmpty(search.ExcludeFieldLabel(search.CVEID))
+	if err != nil {
+		return false, err
+	}
+
+	// check scoping, add as conjunction if needed
+	scope, ok := scoped.GetScope(ctx)
+	if !ok || scope.Level != v1.SearchCategory_IMAGE_VULNERABILITIES {
+		query = search.ConjunctionQuery(query, resolver.getImageCVEQuery())
+	}
+
+	query = search.ConjunctionQuery(query, search.NewQueryBuilder().AddBools(search.Fixable, true).ProtoQuery())
```

Comment:
To avoid deep unwrapping because multiple levels of conjunctions, do this way:

```
conjuncts := []*v1.Query{query, search.NewQueryBuilder().AddBools(search.Fixable, true).ProtoQuery()}
if scope, ok := scoped.GetScope(ctx); !ok || scope.Level != v1.SearchCategory_IMAGE_VULNERABILITIES {
  conjuncts = append(conjuncts, resolver.getImageCVEQuery())
}
query = search.ConjunctionQuery(conjuncts...)
```